python/core/auto_generated/qgstemporalnavigationobject.sip.in

```diff
@@ -109,22 +109,44 @@ Returns the current frame number.
 .. seealso:: :py:func:`setCurrentFrameNumber`
 %End
 
-    void setFrameDuration( QgsInterval duration );
```

Comment:
Are you able to find an approach which doesn't involving changing these? They are part of the stable api guarantee now, so we can't change them.

Reply:
Just thinking out loud here: should in the dropdown of the controllerwidget maybe from 'milliseconds' untill/including 'weeks' use the setFrameDuration, and from 'months' untill/including 'centuries' use the exact setFrameTimeStep?

Would that be an option?

Although this is a hard definition problem: what if you have your frames going from 15th of januari, adding one (exact) month, would that be 31 days, or 28 days? OR would that be exactly 15th of februari? Mmm, probably the last I would think now...

Reply:
@rduivenvoorde That UI suggestion makes sense to me. You could always select 30 days rather than 1 month if you wanted to keep the current ("average") behaviour. Using QDateTime::AddMonths will take you from 15th January to 15 February, which to me is the behaviour that makes the most sense.

I'd still have to add a new method to allow using "exact" / calendar months since like @nyalldawson mentioned, the QgsInterval method is part of the stable API.

If this makes sense, I'd be happy to make those changes and put this back up for review.

Reply:
@nyalldawson what do you think of this? Would this be an issue somewhere else?